src/gg20/crypto_tools/paillier/zk/mta.rs

```diff
@@ -240,7 +238,7 @@ impl ZkSetup {
         let s2 = Randomness(e * &rho.0 + &rho_prime.0);
 
         // t1 = e y + gamma
-        let t1 = Plaintext(e * &wit.msg.0 + &gamma.0); // TODO: This exceeds the modulus N
```

Comment:
I'm not so much worried about length attacks---as you say, we're enforcing a sanity length limit.  I'm more worried about the fact that two different plaintexts map to the same ciphertext.  But I guess in our case it's okay because there's only one possible `t1` that will satisfy the verification check or something?

In an ideal world the `Plaintext` type would enforce the constraint that the value is always less than the Paillier modulus `N`, but unfortunately this modulus varies with the encryption key so a given plaintext might be valid for one key but invalid for another.

Perhaps a better compromise is as follows
* Reinstate the debug assert that the plaintext is less than the modulus.  (We might even consider making this an error, but I guess that would force us to make `encrypt` return a non-fatal `Result` or `Option`.)
* Add a `to_plaintext(x: BigNumber) -> Plaintext`  method to `EncryptionKey` that reduces `x` mod `N` and wraps it in a `Plaintext`.  This way users can ensure they'll never fail the assert (or get an error) if they call `ek.to_plaintext` before `ek.encrypt`.
* Make the type of `t1` a `BigNumber` so that we're forced to explicitly convert it to plaintext (ie. reduce mod N) before encrypting.